<?xml version="1.0" encoding="utf-8"?>
<launch>
	<master auto="start"/>

	<arg name="dashboard" default="true"/>
	<arg name="rviz" default="true"/>
  <arg name="gmapper" default="true"/>
	<arg name="statistics" default="true"/>

	<!-- Set static tf transformations -->
  <group unless="$(arg gmapper)">
	<node pkg="tf" type="static_transform_publisher" name="map_broadcaster" args="0 0 0 0 0 0 1 map odom 10"/>
  </group>
	<param name="robot_description" textfile="$(find simulation)/data/urdf_model/mapping_model.urdf"/>
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
	<!-- Run the necessary nodes to bring up the car -->
	<node pkg="simulation" type="simulation_node" name="simulation" />
	<node pkg="simulation" type="scan_node" name="scan_simulation" output="screen"/>
	<node pkg="simulation" type="usrange_node" name="usrange_simulation" output="screen"/>

  <!-- Run the mapping -->
  <group if="$(arg gmapper)">
    <!-- <include file="$(find simulation)/launch/amcl.launch"/> -->
  	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
	    <param name="scan" value="scan" />
      <param name="maxUrange" value="9.8" />
      <param name="maxRange" value="10.7" />
      <param name="particles" value="120" />
      <param name="srr" value="0.0" />
      <param name="srt" value="0.0" />
      <param name="stt" value="0.0" />
      <param name="str" value="0.0" />
      <param name="lsigma" value="0.0" />
      <param name="minimumScore" value="1000" />
      <param name="map_update_interval" value="0.2" />
			<param name="linearUpdate" value="0.05"/>
  		<param name="angularUpdate" value="0.04"/>
			<param name="iterations" value="0"/>
			<param name="temporalUpdate" value="-1"/>

  	</node>
  </group>

	<!-- Run the mapping -->
  <group if="$(arg statistics)">
  	<node pkg="simulation" type="measurements_node" name="simulation_measurements"/>
  </group>

	<!-- Show the Rviz visualization 	-->
	<group if="$(arg rviz)">
		<node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulation)/data/viewer/viewer.rviz"/>

	</group>
	<!-- Run the Dashboard Application -->
	<group if="$(arg dashboard)">
		<node pkg="simulation" type="dashboard_node" name="dashboard_simulation" output="screen"/>
	</group>
</launch>
